tests/test_dimming_actuator.py

```diff
@@ -0,0 +1,138 @@
+    }
+    outputs = {
+        "odp0000": {"pairingID": 256, "value": "0"},
+        "odp0001": {"pairingID": 272, "value": "50"},
+    }
+    parameters = {}
+
+    return DimmingActuator(
+        device_id="ABB70139AF8A",
+        device_name="Device Name",
+        channel_id="ch0000",
+        channel_name="Channel Name",
+        inputs=inputs,
+        outputs=outputs,
+        parameters=parameters,
+        api=mock_api,
+    )
+
+
+@pytest.mark.asyncio
+async def test_initial_state(dimming_actuator):
+    """Test the intial state of the DimmingActuator."""
+    assert dimming_actuator.state is False
+
+
+@pytest.mark.asyncio
+async def test_turn_on(dimming_actuator):
+    """Test to turning on the DimmingActuator."""
+    await dimming_actuator.turn_on()
+    assert dimming_actuator.state is True
+    dimming_actuator._api.set_datapoint.assert_called_with(
+        device_id="ABB70139AF8A",
+        channel_id="ch0000",
+        datapoint="idp0000",
+        value="1",
+    )
+
+
+@pytest.mark.asyncio
+async def test_turn_off(dimming_actuator):
+    """Test to turning off the DimmingActuator."""
+    await dimming_actuator.turn_off()
+    assert dimming_actuator.state is False
+    dimming_actuator._api.set_datapoint.assert_called_with(
+        device_id="ABB70139AF8A",
+        channel_id="ch0000",
+        datapoint="idp0000",
+        value="0",
+    )
+
+
+@pytest.mark.asyncio
+async def test_set_brightness(dimming_actuator):
+    """Test to set brightness off the DimmingActuator."""
+    await dimming_actuator.set_brightness(50)
+    assert dimming_actuator.brightness == 50
+    await dimming_actuator.set_brightness(-1)
+    assert dimming_actuator.brightness == 1
+    await dimming_actuator.set_brightness(110)
+    assert dimming_actuator.brightness == 100
+
+
+@pytest.mark.asyncio
+async def test_refresh_state(dimming_actuator):
+    """Test refreshing the state of the DimmingActuator."""
+    dimming_actuator._api.get_datapoint.return_value = ["1"]
+    await dimming_actuator.refresh_state()
+    assert dimming_actuator.state is True
+    dimming_actuator._api.get_datapoint.assert_called_with(
+        device_id="ABB70139AF8A",
+        channel_id="ch0000",
+        datapoint="odp0001",
+    )
+
+
+def test_refresh_state_from_output(dimming_actuator):
+    """Test the _refresh_state_from_output function."""
+    # Check output that affects the state.
+    dimming_actuator._refresh_state_from_output(
+        output={"pairingID": 256, "value": "1"},
+    )
+    assert dimming_actuator.state is True
+
+    # Check output that affects the brightness
+    dimming_actuator._refresh_state_from_output(
+        output={"pairingID": 272, "value": 75},
+    )
+    assert dimming_actuator.brightness == 75
+
+    # Check output that does NOT affect the state.
+    dimming_actuator._refresh_state_from_output(
+        output={"pairingID": 256, "value": "0"},
+    )
+    assert dimming_actuator.state is False
```

Comment:
Refresh the state from an output that doesn't affect the state.

```suggestion
    # Check output that does NOT affect the state,
    # ensure state, brightness are unchanged.
    dimming_actuator._refresh_state_from_output(
        output={"pairingID": 257, "value": "0"},
    )
    assert dimming_actuator.brightness == 75
    assert dimming_actuator.state is True
```